tap/tlstapper/bpf/common.c

```diff
@@ -104,6 +104,8 @@ static __always_inline void output_ssl_chunk(struct pt_regs *ctx, struct ssl_inf
     chunk->len = count_bytes;
     chunk->fd = info->fd;
 
+    (void)memcpy(&chunk->kprobe_address_pair, &info->kprobe_address_pair, sizeof(chunk->kprobe_address_pair));
```

Comment:
I wonder why not to use the `bpf_probe_read` which is the both the standard method in eBPF and used in our code base.

Indeed loops in eBPF are limited, new Kernels allows bounded loops (loop from X to known and small Y). While older kernels don't allow it and force the use of Unrolling loops, which is basically a copy paste of the loop body.
I assume the compiler's built it `__builtin_memcpy` do small bounded loop so new kernels allow it, but its better to use the `bpf_probe_read` which does the same and supported by all kernel versions.

Reply:
Fixed in a4948dbd9f8d44355b997ac7e2efcb5eae4e5dd0.

I thought that `bpf_probe_read` is meant for reading from kernel space, and therefore it would be confusing to see it used otherwise. (for example, here it might imply to the reader that `chunk->kprobe_address_pair` is in kernel space. It is just a matter of convention. If it is used in our code as a replacement for memcpy then should be used here as well.